did_doc_sov/src/lib.rs

```diff
@@ -0,0 +1,117 @@
+pub mod error;
+pub mod extra_fields;
+pub mod service;
+
+use did_doc::{
+    did_parser::{Did, DidUrl},
+    schema::{
+        did_doc::{ControllerAlias, DidDocument, DidDocumentBuilder},
+        types::uri::Uri,
+        verification_method::{VerificationMethod, VerificationMethodKind},
+    },
+};
+use error::DidDocumentSovError;
+use extra_fields::ExtraFieldsSov;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use service::{services_list::ServicesList, ServiceSov};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct DidDocumentSov {
+    #[serde(flatten)]
+    did_doc: DidDocument<ExtraFieldsSov>,
+}
+
+impl DidDocumentSov {
+    pub fn builder() -> DidDocumentSovBuilder {
+        DidDocumentSovBuilder::default()
+    }
+
+    pub fn id(&self) -> &Did {
+        self.did_doc.id()
+    }
+
+    pub fn also_known_as(&self) -> &[Uri] {
+        self.did_doc.also_known_as()
+    }
+
+    pub fn controller(&self) -> Option<&ControllerAlias> {
+        self.did_doc.controller()
+    }
+
+    pub fn verification_method(&self) -> &[VerificationMethod] {
+        self.did_doc.verification_method()
+    }
+
+    pub fn authentication(&self) -> &[VerificationMethodKind] {
+        self.did_doc.authentication()
+    }
+
+    pub fn service(&self) -> Result<ServicesList, DidDocumentSovError> {
+        Ok(ServicesList::new(
+            self.did_doc
+                .service()
+                .iter()
+                .map(|s| ServiceSov::try_from(s.to_owned()))
+                .collect::<Result<Vec<ServiceSov>, _>>()?,
+        ))
```

Comment:
TODO: The conversion should be be done once during deserialization / building of `DidDocumentSov`.